integrate/trapezoidal_test.go

```diff
@@ -11,6 +11,15 @@ import (
 	"gonum.org/v1/gonum/floats"
 )
 
+func assertPanic(t *testing.T, f func() float64, desiredPanic string) {
```

Comment:
The idiom we use for this is not an assert, but rather a function that will returns whether a function call panicked, and what message the panic was, like so https://github.com/gonum/gonum/blob/287df04a63144ae62d3f7a85dc313836f665c85b/mat/matrix_test.go#L18-L26

Reply:
Looks like `panics` is present in several different test files in very similar forms.  Would you advise putting a single copy of the function under `gonum/internal`?

Reply:
Yeah, maybe. If we make a test helpers package in internal that might be worthwhile, but I think that should be a separate PR, with a proposal issue first.

Reply:
Agreed.